lib/SILOptimizer/Mandatory/TFPartition.cpp

```diff
@@ -317,6 +321,15 @@ static bool isAcceleratorOnly(const SILFunction &hostFn) {
          SILFunctionType::Representation::TensorFlow; // @convention(tensorflow)
 }
 
+/// Return true if a user instruction is returning or forming a return value.
+static bool isReturning(SILInstruction *user) {
+  if (isa<ReturnInst>(user)) return true;
+  if (auto *TI = dyn_cast<TupleInst>(user))
```

Comment:
in the new world of graph_op inst (a multi-value inst type), i believe we don't need to deal with returning a TupleInst any more. e.g. see https://github.com/apple/swift/blob/8b3b1726ffd2bc6e810fcb066838f6906881b6c6/lib/SILOptimizer/Mandatory/TFPartition.cpp#L4236

Reply:
I don't think this is unrelated. The result here is the result of a `graph_op`, but it is not the result of `hostFn`. Returns to `hostFn` still follow the SIL function return rule. That is, if the `hostFn` has multiple results, we need to form and return a tuple to abide by SIL semantics.

For example:
```
sil @foo : $(...) -> (A, B) {
   %a = ... (has type $A)
   %b = ... (has type $B)
   %ret = tuple (%a, %b)
   return %ret
}
```

Reply:
You are right. Perhaps rename this func name to be more specific, like `isReturningFromHostFunc`?

Reply:
I'm indifferent, but what's the benefit of specializing it for "host func" in the name? 

The logic is independent from partitioning, it's just checking whether an instruction is returning or preparing for returning in a general way. The logic applies to instructions in any function, not just `hostFn`.

Reply:
I don't feel strongly about the naming either. 

should we add a unit test on the tuple case?

Reply:
Added.

Reply:
Tuple test is failing due to a crasher: [SR-8506](https://bugs.swift.org/browse/SR-8506). I'll add the test later.

Non-tuples are already being tested in dataset_1.swift.